external/kafka-0-10-sql/src/test/scala/org/apache/spark/sql/kafka010/KafkaMicroBatchSourceSuite.scala

```diff
@@ -204,6 +204,43 @@ abstract class KafkaMicroBatchSourceSuiteBase extends KafkaSourceSuiteBase {
       StopStream)
   }
 
+  test("SPARK-26718 Rate limit set to Long.Max should not overflow integer " +
+    "during end offset calculation") {
+    val topic = newTopic()
+    testUtils.createTopic(topic, partitions = 1)
+    // fill in 5000 messages to trigger potential integer overflow
+    testUtils.sendMessages(topic, (0 until 5000).map(_.toString).toArray, Some(0))
+
+    val partitionOffsets = Map(
+      new TopicPartition(topic, 0) -> 5000L
+    )
+    val startingOffsets = JsonUtils.partitionOffsets(partitionOffsets)
+
+    val sparkSession = spark
+    import sparkSession.implicits._
```

Comment:
Are the above two lines required? Shall we remove these?
```scala
-    val sparkSession = spark
-    import sparkSession.implicits._
```

Reply:
ha, it's not, Encoders are already provided in this context as implicits, I will remove it.